src/browser/provider/module-loader.js

```diff
@@ -0,0 +1,24 @@
+const BROWSER_PROVIDER_SCOPED_PACKAGE_RE = /^@\S+\/\S/;
+const BROWSER_PROVIDER_PARSER_RE = /@([\w-]+)\/(.*)/g;
+
+export default class BrowserProviderModuleLoader {
```

Comment:
I feel like that functionality of this function is too simple for creating a class. Actually, it can be accomplished just by the one function.

Let's just turn th `loadModule` method to the `getProviderModuleName` function, and make it the default export. Also, make it return just a provider name instead of a provider object, and do `require(getProviderModuleName(providerName))` call in the `BrowserProviderPool` class.

In this case, the file should be named as `get-provider-module-name.js`.

Reply:
@AndreyBelym This is more challenging for me - the problem (and the reason why I refactored the code out to a class) is that I want to test only the behaviour of the conversion of the argument from the command line into the package name that I wish to load. Particularly I need to test the use case that the existing behaviour (public non-scoped packages) works in regression as well as enabling checks to handle scoped packages are behaving correctly without attempting to load the actual package (which is both slow from a test performance perspective and requires me to valid, installed packages to require()). By merging the detection of the module scoping with the require()ing of packages we merge those two responsibilities, which then makes the code untestable. 

I'm happy to do so, however this means that I can't see how we could write tests to confirm that behaviour short of client-side tests that are overweight? 

Of course, I'm happy to follow your direction and I'm still learning this whole JS thing so I'm happy to be enlightened.. :-)